core/cas-server-core-services/src/main/java/org/apereo/cas/services/DefaultRegisteredServiceAccessStrategy.java

```diff
@@ -230,91 +230,63 @@ public boolean doPrincipalAttributesAllowServiceAccess(final String principal, f
     private boolean doRequiredAttributesAllowPrincipalAccess(final Map<String, Object> principalAttributes) {
         LOGGER.debug("These required attributes [{}] are examined against [{}] before service can proceed.",
                 this.requiredAttributes, principalAttributes);
-        final Sets.SetView<String> difference = Sets.intersection(this.requiredAttributes.keySet(), principalAttributes.keySet());
-        final Set<String> copy = difference.immutableCopy();
+        final Set<String> difference = this.requiredAttributes.keySet().stream()
```

Comment:
I can change if you want to, but I, personally, don't like the way you have to use ```ratainAll```:
```
final Set<String> difference = new HashSet<>(this.requiredAttributes.keySet());
        difference.retainAll(principalAttributes.keySet());
```
It looks like 2 separate operations, when it's only 1, for me stream is much clearer